android/PartialSync/app/src/main/java/io/realm/todo/WelcomeActivity.java

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+
+package io.realm.todo;
+
+import android.animation.Animator;
+import android.animation.AnimatorListenerAdapter;
+import android.annotation.TargetApi;
+import android.content.Intent;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+
+import io.realm.ObjectServerError;
+import io.realm.Realm;
+import io.realm.SyncConfiguration;
+import io.realm.SyncCredentials;
+import io.realm.SyncUser;
+
+import static io.realm.todo.Constants.AUTH_URL;
+import static io.realm.todo.Constants.REALM_URL;
+
+public class WelcomeActivity extends AppCompatActivity {
+
+    private EditText mNicknameTextView;
+    private View mProgressView;
+    private View mLoginFormView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_welcome);
+
+        if (SyncUser.currentUser() != null) {
+            setUpRealmAndGoToListTaskActivity();
+        }
+
+        // Set up the login form.
+        mNicknameTextView = findViewById(R.id.nickname);
+        Button loginButton = findViewById(R.id.login_button);
+        loginButton.setOnClickListener(view -> attemptLogin());
+        mLoginFormView = findViewById(R.id.login_form);
+        mProgressView = findViewById(R.id.login_progress);
+    }
+
+    private void attemptLogin() {
+        // Reset errors.
+        mNicknameTextView.setError(null);
+        // Store values at the time of the login attempt.
+        String nickname = mNicknameTextView.getText().toString();
+        showProgress(true);
+
+        SyncCredentials credentials = SyncCredentials.nickname(nickname, true);
+        SyncUser.loginAsync(credentials, AUTH_URL, new SyncUser.Callback<SyncUser>() {
+            @Override
+            public void onSuccess(SyncUser user) {
+                showProgress(false);
+                setUpRealmAndGoToListTaskActivity();
+            }
+
+            @Override
+            public void onError(ObjectServerError error) {
+                showProgress(false);
+                mNicknameTextView.setError("Uh oh something went wrong! (check your logcat please)");
+                mNicknameTextView.requestFocus();
+                Log.e("Login error", error.toString());
+            }
+        });
+    }
+
+    /**
+     * Shows the progress UI and hides the login form.
+     */
+    @TargetApi(Build.VERSION_CODES.HONEYCOMB_MR2)
+    private void showProgress(final boolean show) {
+        int shortAnimTime = getResources().getInteger(android.R.integer.config_shortAnimTime);
+        mLoginFormView.setVisibility(show ? View.GONE : View.VISIBLE);
+        mLoginFormView.animate().setDuration(shortAnimTime).alpha(
+                show ? 0 : 1).setListener(new AnimatorListenerAdapter() {
```

Comment:
Convert to lambda shorthand syntax?  Otherwise you are mixing styles in this file and others

Reply:
#LegacyCode, also `setListener` doesn't accept a Functional Interface. so you can't use lambda here AFAIK